app/src/main/java/com/example/una/adapters/StreaksComplexRecyclerViewAdapter.java

```diff
@@ -0,0 +1,82 @@
+package com.example.una.adapters;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.una.Challenge;
+import com.example.una.R;
+import com.example.una.Viewholders.ChallengeViewHolder;
+import com.example.una.Viewholders.StreaksViewHolder;
+
+import java.util.List;
+
+public class StreaksComplexRecyclerViewAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+    // The challenges to display in your RecyclerView
+    private List<Object> items;
+    private final int IMAGE = 0, CHALLENGE = 1;
+
+    public StreaksComplexRecyclerViewAdapter(List<Object> items) {
+        this.items = items;
+    }
+
+    @Override
+    public int getItemCount() {
+        return this.items.size();
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        if (position == 0) {
+            return IMAGE;
+        } else {
+            return CHALLENGE;
+        }
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {
+        RecyclerView.ViewHolder viewHolder;
+        LayoutInflater inflater = LayoutInflater.from(viewGroup.getContext());
+        switch (viewType) {
+            case CHALLENGE:
+                View itemViewChallenge = inflater.inflate(R.layout.challenge_viewholder_layout, viewGroup, false);
+                viewHolder = new ChallengeViewHolder(itemViewChallenge);
+                break;
+            case IMAGE:
+                View itemViewStreaks = inflater.inflate(R.layout.streaks_viewholder_layout, viewGroup, false);
+                viewHolder = new StreaksViewHolder(itemViewStreaks);
+                break;
+            default:
+                View v = inflater.inflate(android.R.layout.simple_list_item_1, viewGroup, false);
+                viewHolder = new StreaksViewHolder(v);
+                break;
+        }
+        return viewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, int position) {
+        if (position == 0) {
+            StreaksViewHolder streaksViewHolder = (StreaksViewHolder) viewHolder;
+            configureStreakViewHolder(streaksViewHolder);
+        } else {
+            ChallengeViewHolder challengeViewHolder = (ChallengeViewHolder) viewHolder;
+            configureChallengeViewHolder(challengeViewHolder, position);
+        }
+    }
+
+    private void configureChallengeViewHolder(ChallengeViewHolder vhChallenge, int position) {
```

Comment:
I like how you separated these into their own methods instead of cluttering the `onBindViewHolder`.